v4/core/request_builder.go

```diff
@@ -305,6 +314,17 @@ func (requestBuilder *RequestBuilder) Build() (*http.Request, error) {
 		}
 	}
 
+	// If we have a request body and gzip is enabled, then wrap the body in a Gzip compression reader
+	// and add the "Content-Encoding: gzip" request header.
+	if !IsNil(requestBuilder.Body) && requestBuilder.EnableGzipCompression {
```

Comment:
We should add a check to make sure the `Content-Encoding` header was not already set: https://github.com/IBM/java-sdk-core/blob/master/src/main/java/com/ibm/cloud/sdk/core/http/gzip/GzipRequestInterceptor.java#L31

Reply:
Actually, I don't think that's the logic that we want, tbh...   We should discuss.

Reply:
To expand on this a bit...  In general, HTTP headers can have multiple values.  Content-Encoding is perhaps one of those headers that could conceivably have multiple values because it's possible that the payload has had multiple "encodings" performed on it.   Perhaps our check should be that the list of values returned for the Content-Encoding header does not contain "gzip".

Reply:
okay, if that needs to be updated in java then I can make that change.